Removed $.browser

gseguin opened the pull request.

Commit:
Removed use of $.browser

jazzfog commented:
Are you going to fix it? jQuery 1.9 is here.

pehrlich commented:
+1

ghost commented:
Hi,

We have updated our site with jQuery 1.9 and the hashchange plugin is the only one not working! It triggers an error, please see here: https://github.com/cowboy/jquery-hashchange/issues/25

Please is there a quick fix, I was thinking of an alternative to detect IE? 

We've removed the hashchange plugin altogether for now and are actively looking at an alternative or a fix... Our entire site now suffers from missing back button support, what a shame :(
Thanks for any help!

pehrlich commented:
@anonumus As per the upgrade guide, use this: https://github.com/jquery/jquery-migrate/

JohnCz commented:
a quick and dirty fix would be to replace the one instance of ...

   $.browser.msie

with

   /msie/.test(navigator.userAgent.toLowerCase())

... this is by no means full proof

jazzfog commented:
My Solution is

Open source plugin source code and replace 

``` javascript
$.browser.msie
```

 with

``` javascript
(navigator.appName == 'Microsoft Internet Explorer')
```

fdaugan commented:
Better solution :)

``` javascript
(navigator.appName === 'Microsoft Internet Explorer')
```

jazzfog commented:
Is there any difference in this particular case? ;-)

ghost commented:
Thanks all for the quick fixes ;)
Using jQuery Migrate isn't an option... we want to move forward, not backward, we have 10 others libs/plugins that work great with jQuery 1.9, this one is the oldest one without an update in 2 years.. and is the only one that triggers an error!

By the way, I'm surprised that there is no other option than this outdated plugin for an "hashchange event" in jquery, considering the popularity of sites using ajax that use URLs like /#!pageID (i.e. twitter & co...) Am I missing something? or is there another way to detect the "hashchange" event in jQuery? or is this really THE ONLY plugin that exists for this????

fdaugan commented:
You can use "routie" : no JQuery dependency, but works only for IE8+ and modern browsers. I was used to use routie before I had to support IE7.

pthorson commented:
@fabdouglas Did you already to look at https://github.com/andreasbernhard/history.js ?

fdaugan commented:
History.js does not handle direct links such as :

``` html
<a href="www.mysite.com/#?state=1&_suid=1">Display</a>
```

It manipulates history, so covers another web feature. (personnaly) I use jquery-hashchange to listen this internal navigation requests to load some content. This way I cover bookmarks, back, forward, etc.

ghost commented:
I just discovered History.js and it looks really nice!! I'm seriously considering using it! However the main concern is its huge weight, come on 20KB+ minified for one simple feature... As a comparison, jquery.ba-haschange is 2KB only... I'd rather keep with it and implement the HTML5 history stuff myself...

danbalbo commented:
We have removed the browser sniffing condition: the plugin already checks, in a better way, if browser supports the hashchange event:  
supports_onhashchange = 'on' + str_hashchange in window && ( doc_mode === undefined || doc_mode > 7 );

So we changed this line:
$.browser.msie && !supports_onhashchange && (function(){
in:
!supports_onhashchange && (function(){

sylvainlg commented:
Another way to fix this bug http://stackoverflow.com/questions/14512826/impromptu-with-jquery-1-9-error-with-browser-msie

Tested with Jquery 1.9.1

andey commented:
Change line 300

From:

```
$.browser.msie && !supports_onhashchange && (function(){
```

To:

```
/MSIE/.test(navigator.userAgent) && !supports_onhashchange && (function(){
```

That's all jQuery.browser does anyways.

> Description: Contains flags for the useragent, read from navigator.userAgent. We recommend against using this property; please try to use feature detection instead (see jQuery.support). jQuery.browser may be moved to a plugin in a future release of jQuery.

http://api.jquery.com/jQuery.browser/

rkingon commented:
I kind of prefer not to modify plugins directly... If you do something along these lines before you include hashchange plugin that would work as well...

$.browser = {msie: (navigator.appName === "Microsoft Internet Explorer")};

not tested in IE, just thought of it from the comments above- this is assuming the fact that navigator.appName is a safe way to test for IE (yes, we know nothing is a safe way to test)

Commit:
Make pasting inline into <script> tag possible

The browser interprets the occurrence of the literal "</script>" inside of the
body of a script tag as the end of the script, so the leading "<" must be
escaped by replacing it with \x3c
